lib/std/comptime_string_map.zig

```diff
@@ -50,26 +51,49 @@ pub fn ComptimeStringMap(comptime V: type, comptime kvs: var) type {
     };
 
     return struct {
-        pub fn has(str: []const u8) bool {
-            return get(str) != null;
-        }
-
-        pub fn get(str: []const u8) ?V {
+        pub fn getWithEqlFn(str: []const u8, eql: fn (a: []const u8, b: []const u8) bool) ?V {
             if (str.len < precomputed.min_len or str.len > precomputed.max_len)
                 return null;
 
             var i = precomputed.len_indexes[str.len];
             while (true) {
                 const kv = precomputed.sorted_kvs[i];
-                if (kv.key.len != str.len)
-                    return null;
```

Comment:
Removing this early return removes a big part of what makes ComptimeStringMap fast--that is, it was only comparing keys of equal length, and is able to do that by sorting keys by length at comptime.

To give a concrete example, for a map with keys "a", "ab", "abc", "abcd":

- Before: `get("xy")` would start checking for equality with "ab", next iteration would be against "abc", but len would not match, so it'd return `null` early (i.e. it'd only check equality once)
- Now: `get("xy")` would start checking for equality with "ab", then it'd check equality with "abc", then "abcd" and finally return `null` once it reaches the end of `sorted_kvs`

Without this early return optimization, this would only be partially better than a naive array lookup on average.

Reply:
> Removing this early return removes a big part of what makes ComptimeStringMap fast--that is, it was only comparing keys of equal length, and is able to do that by sorting keys by length at comptime.

See `ascii.eql` in this PR.

Reply:
That doesn't matter, returning `null` early (from inside the loop) is the optimization, not the length comparison itself. With the implementation in this PR, the loop will continue checking keys with unequal lengths. Before this PR, the loop would return `null` once lengths were unequal.

Benchmarking the Zig tokenizer with this PR should show a significant performance decrease.

Reply:
Oh, I see. Thanks! :)